src/dotty/tools/dotc/util/Util.scala

```diff
@@ -0,0 +1,35 @@
+package dotty.tools.dotc.util
+import reflect.ClassTag
+
+object Util {
+  
+  /** The index `i` in `candidates.indices` such that `candidates(i) <= x` and
+   *  `candidates(i)` is closest to `x`, determined by binary search, or -1
+   *  if `x < candidates(0)`.
+   *  @param  hint   If between 0 and `candidates.length` use this
+   *                 as the first search point, otherwise use 
+   *                 `candidates.length/2`.
+   *  @pre   candidates is sorted
+   *  @pre   candidates(0) <= x
```

Comment:
The second precondition should not be there, since you specify what happens if it's not true, i.e., the method must return -1. (as opposed to: it crashes or loops indefinitely)